app/assets/javascripts/avalon.js

```diff
@@ -14,101 +14,112 @@
+     */
+    if (btn.classList.contains('btn-stateful-loading') && btn?.dataset?.loadingText != undefined) {
+      btn.value = btn.dataset.loadingText;
+      btn.style.cursor = 'not-allowed';
+      btn.style.opacity = 0.5;
+    }
   });
 
-  var iOS = !!/(iPad|iPhone|iPod)/g.test(navigator.userAgent);
-  if (iOS) {
-    $('input[readonly], textarea[readonly]').on('cut paste keydown', function (
-      e
-    ) {
-      e.preventDefault();
-      e.stopPropagation();
-      return false;
+  const showObjectTree = getById('show_object_tree');
+  if (showObjectTree) {
+    showObjectTree.addEventListener('click', function () {
+      const objectTree = getById('object_tree');
+      if (objectTree) {
+        fetch(objectTree.dataset.src)
+          .then(response => response.text())
+          .then(html => {
+            objectTree.innerHTML = html;
+          });
+      }
     });
-    $('input[readonly], textarea[readonly]').attr('readonly', false);
   }
 
-  $('a')
-    .has('img, ul')
-    .addClass('block');
+  const iOS = !!/(iPad|iPhone|iPod)/g.test(navigator.userAgent);
+  function preventDefaultAndStop(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    return false;
+  }
+  if (iOS) {
+    const readonlyInputs = queryAll('input[readonly], textarea[readonly]');
+    readonlyInputs.forEach(function (input) {
+      input.addEventListener('cut', preventDefaultAndStop);
+      input.addEventListener('paste', preventDefaultAndStop);
+      input.addEventListener('keydown', preventDefaultAndStop);
+      input.removeAttribute('readonly');
+    });
+  }
 
-  window.addEventListener(
-    'hashchange',
-    function (event) {
-      var element = document.getElementById(location.hash.substring(1));
-      if (element) {
-        if (!/^(?:a|select|input|button|textarea)$/i.test(element.tagName)) {
-          element.tabIndex = -1;
-        }
-        element.focus();
+  window.addEventListener('hashchange', function () {
+    var element = getById(location.hash.substring(1));
+    if (element) {
+      if (!/^(?:a|select|input|button|textarea)$/i.test(element.tagName)) {
+        element.tabIndex = -1;
       }
-    },
-    false
-  );
+      element.focus();
+    }
+  }, false);
 
   // Set CSS to push the page content above footer
-  $('.content-wrapper').css('padding-bottom', $('#footer').css('height'));
+  const contentWrapper = query('.content-wrapper');
+  const footer = getById('footer');
 
-  // Re-set the space between content-wrapper and footer on window resize.
-  // With this the main content height is adjusted when orientation changes when
-  // using mobile devices, avoiding main content bleeding into the footer.
-  window.addEventListener("resize", () => {
-    $('.content-wrapper').css('padding-bottom', $('#footer').css('height'));
-  }, true);
+  window.addEventListener('resize', () => {
+    if (contentWrapper && footer) { }
+  });
```

Comment:
It looks like this can be removed since I think this was supposed to call `adjustFooterPadding` and it is handled in the `resize` listener below.